hyperspy/component.py

```diff
@@ -756,6 +756,9 @@ def summary(self):
                                                parameter.std,
                                                parameter.units)
 
+    def __tempcall2d__(self, p, x, y, onlyfree=True):
+        self.fetch_values_from_array(p, onlyfree=onlyfree)
+        return self.function(x, y)
```

Comment:
...though I just may be avoiding rewriting the parameters to be truly multi-dimensional. It's (apparently) the oldest part of the HyperSpy, it may be time to rewrite / restructure the code. It's a big task, though. I'd love to do it, but don't think I can afford to atm :)